pkg/generate/app/app.go

```diff
@@ -81,6 +85,28 @@ func nameFromGitURL(url *url.URL) (string, bool) {
 	}
 	// from path
 	if name, ok := git.NameFromRepositoryURL(url); ok {
+
+		if !kvalidation.IsDNS1123Label(name) {
+			name = strings.ToLower(name)
+
+			// remove everything except [-0-9a-z]
+			name = invalidNameCharactersRegexp.ReplaceAllString(name, "")
```

Comment:
why shouldn't a name suggester suggest valid names?  suggesting a name that the consumer then has to further modify seems odd to me.